### PR TITLE
expose prql as lib + integration test

### PR DESCRIPTION
**Pull request by kwigley**
…yaml

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -11,7 +11,6 @@ repos:
     rev: v1.0.1
     hooks:
       - id: fmt
-      - id: check
       - id: clippy
```
Comment (kwigley) [inline .pre-commit-config.yaml:14]: clippy runs `check` under the hood 👍

File: tests/test_parser.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+use prql::{ast_of_string, Result, Rule};
```
Comment (kwigley) [inline tests/test_parser.rs:1]: test prql as a lib

File: src/cli.rs [hunk 3]
```diff
@@ -27,11 +27,9 @@ impl Cli {
         let mut source = String::new();
         self.input.read_to_string(&mut source)?;
         match self.format {
-            Dialect::Sql => writeln!(
-                self.output,
-                "TODO! do something useful with source:\n{}",
-                &source
-            )?,
+            Dialect::Ast => self
```
Comment (max-sixty) [inline src/cli.rs:30]: Oh clever, I had been thinking of it as a different subcommand but this is better!

File: tests/test_parser.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+#[test]
+fn parse_simple_string_to_ast() -> Result<()> {
+    assert_eq!(
+        serde_yaml::to_string(&ast_of_string("select 1", Rule::query)?)?,
```
Comment (max-sixty) [inline tests/test_parser.rs:6]: Very nice!

FYI if we use `insta`, we get the value added in "for free" (but no need to change)

**Review by max-sixty — APPROVED** [review]
Great, thanks @kwigley !

**Comment by max-sixty** [comment]
Very cool!

```
$ echo "from employees
    select x
    sort y" |
      ./target/debug/prql
---
Query:
  - Pipeline:
      - From:
          - Ident: employees
      - Select:
          - Ident: x
      - Sort:
          - Ident: y
```

The SQL output got _worse_ this weekend given I've been changing how functions run, so I'll finish that off this week and then we'll have this running for SQL, at least for some limited set of queries — the first two examples is my goal

**Pull request merged**